include/seqan3/range/views/minimiser_hash.hpp

```diff
@@ -0,0 +1,158 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Mitra Darvish <mitra.darvish AT fu-berlin.de>
+ * \brief Provides seqan3::views::minimiser_hash.
+ */
+
+#pragma once
+
+#include <seqan3/range/views/kmer_hash.hpp>
+#include <seqan3/range/views/minimiser.hpp>
+
+namespace seqan3::detail
+{
+//!\brief views::minimiser_hash's range adaptor object type (non-closure).
+struct minimiser_hash_fn
+{
+    /*!\brief Store the shape and the window size and return a range adaptor closure object.
+    * \param[in] shape       The seqan3::shape to use for hashing.
+    * \param[in] window_size The windows size to use.
+    * \throws std::invalid_argument if the size of the shape is greater than the `window_size`.
+    * \returns               A range of converted elements.
+    */
+    constexpr auto operator()(shape const & shape, uint32_t const window_size) const
+    {
+        return seqan3::detail::adaptor_from_functor{*this, shape, window_size};
+    }
+
+    /*!\brief Store the shape, the window size and the seed and return a range adaptor closure object.
+    * \param[in] shape       The seqan3::shape to use for hashing.
+    * \param[in] window_size The size of the window.
+    * \param[in] seed        The seed to use.
+    * \throws std::invalid_argument if the size of the shape is greater than the `window_size`.
+    * \returns               A range of converted elements.
+    */
+    constexpr auto operator()(shape const & shape, uint32_t const window_size, uint64_t const seed) const
+    {
+        return seqan3::detail::adaptor_from_functor{*this, shape, window_size, seed};
+    }
+
+    /*!\brief            Call the view's constructor with the underlying view, a seqan3::shape and a window size as
+     *                   argument.
+     * \param[in] urange      The input range to process. Must model std::ranges::viewable_range and the reference type
+     *                        of the range must model seqan3::semialphabet.
+     * \param[in] shape       The seqan3::shape to use for hashing.
+     * \param[in] window_size The size of the window.
+     * \param[in] seed        The seed to use.
+     * \throws std::invalid_argument if the size of the shape is greater than the `window_size`.
+     * \returns               A range of converted elements.
+     */
+    template <std::ranges::range urng_t>
+    constexpr auto operator()(urng_t && urange, shape const & shape, uint32_t const window_size,
```

Comment:
if it doesn't fit all on one line, put one parameter on each new line :nail_care: 

```cpp
foo(param1 p1,
    param2 p2,
      // ... 
```